src/Analyzer.Utilities/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -93,8 +93,18 @@ public static IEnumerable<ITypeSymbol> GetBaseTypesAndThis(this ITypeSymbol type
             }
         }
 
-        public static bool DerivesFrom(this INamedTypeSymbol symbol, INamedTypeSymbol candidateBaseType)
+        public static bool DerivesFrom(this ITypeSymbol symbol, ITypeSymbol candidateBaseType, bool baseTypesOnly = false)
```

Comment:
It is clearly the ITypeSymbol interface has a bad design on this AllInterfances property, which is against the DIP OOP principle [(DIP).](https://en.wikipedia.org/wiki/Dependency_inversion_principle) A base interface is depending on a sub interface?? 

You guys should notify somebody to update the design of ITypeSymbol, make the AllInterfances return an array of ITypeSymbol, then it really get generalized.